editor.planx.uk/src/pages/FlowEditor/lib/store/team.ts

```diff
@@ -130,22 +135,20 @@ export const teamStore: StateCreator<
    */
   fetchCurrentTeam: async () => {
     const { teamSlug, $client } = get();
-    const team = await $client.team.getBySlug(teamSlug);
-    return team;
+    return await $client.team.getBySlug(teamSlug);
   },
 
   updateTeamTheme: async (theme: Partial<TeamTheme>) => {
     const { teamId, $client } = get();
-    const isSuccess = await $client.team.updateTheme(teamId, theme);
-    return isSuccess;
+    return await $client.team.updateTheme(teamId, theme);
   },
 
   updateTeamSettings: async (teamSettings: Partial<TeamSettings>) => {
     const { teamId, $client } = get();
-    const isSuccess = await $client.team.updateTeamSettings(
-      teamId,
-      teamSettings,
-    );
-    return isSuccess;
+    return await $client.team.updateTeamSettings(teamId, teamSettings);
+  },
+
+  setTeamMembers: async (teamMembers: TeamMember[]) => {
+    set((state) => ({ teamMembers: [...state.teamMembers, ...teamMembers] }));
```

Comment:
Nice catch! I fixed this then hit an interesting bug where it seemed like the database query to `GetUsersForTeam` wasn't being run when the user went back on the `/members` route. (e.g. Add member, see new member in the table, click on Design then click back to Team members and the new user had disappeared). 

🐰 I went down a whole rabbit hole about query caching and found this great article: https://www.apollographql.com/blog/demystifying-cache-normalization.

But the solution I've ended up with now involves refetching when the mutation completes, which maybe makes the store solution a bit redundant? I suppose it's still useful for what would otherwise be a loading state between mutating and querying, but I'd be interested to hear your thoughts!

Reply:
This is actually a great solution - we don't really leverage the caches in Apollo enough. All working as expected. Agreed - I think this does now make the store approach redundant, it might be worth tidying this up? 

I think if we were making this query just via planx-core, this approach wouldn't work as easily. Something I really want to take a proper look at (soon!) is the number of different ways the app is fetching and managing data - there's room for simplification and standardisation here.